src/components/Header.js

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+
+export default class Header extends React.Component{
+    constructor(props){
+        super(props);
+        this.state = {
+            text: ''
+        };
+    }
+
+    onChange = (e) => {
+        this.setState({
+            text: e.target.value
+        });
+    };
+    onSubmit = (e) => {
+        e.preventDefault();
+        this.props.addItem(this.state.text);
+        this.setState({
+            text: ''
+        })
+    };
+    onKey = (e) => {
```

Comment:
Wrap your input in `<form>`. This way you'll have native submit on enter keypress